### PR TITLE
fix: Delete automation resources in correct order

### PR DESCRIPTION
**Pull request by warber**
#### What this PR does / Why we need it:
This PR adapts the Delete logic so that deletion of automation resources is happening in correct order, Workflows --> Scheduling Rules --> BusinessCalendars.

### CODE REVIEW

File: pkg/delete/delete.go [hunk 1]
```diff
@@ -46,37 +46,56 @@ type DeleteEntries = map[configurationType][]pointer.DeletePointer
...
-			err = bucket.Delete(ctx, clients.Buckets, entries)
 		} else { // assume it's a Settings Schema
 			err = setting.Delete(ctx, clients.Settings, entries)
 		}
-
 		if err != nil {
 			log.WithFields(field.Error(err)).Error("Error during deletion: %v", err)
 			deleteErrors += 1
 		}
+		delete(entriesToDelete, entryType)
```
Comment (UnseenWizzard) [inline pkg/delete/delete.go:96]: is there any need to delete the entries here? If we don't have to, why modify the thing we're looping over already

File: pkg/delete/delete.go [hunk 2]
```diff
@@ -46,37 +46,56 @@ type DeleteEntries = map[configurationType][]pointer.DeletePointer
...
+		if err != nil {
+			log.WithFields(field.Error(err)).Error("Error during deletion: %v", err)
+			deleteErrors += 1
+		}
+		delete(entriesToDelete, key)
+	}
 
-		var err error
+	// Delete bucket resources
+	if reflect.ValueOf(clients.Buckets).IsNil() {
```
Comment (UnseenWizzard) [inline pkg/delete/delete.go]: nit: other than the reflection, and the fact that we can access it directly, is there a reason to handle bucket deletion separate to configs or settings? We could also just delete the bucket entries if no client is defined and leave the delete call in the main loop over delete entries

File: pkg/delete/delete.go (hunk 1 above)
Comment (warber) [reply pkg/delete/delete.go:98]: not really, knowing that the map is empty at the end just has a calming effect :D

File: pkg/delete/delete.go [hunk 3]
```diff
@@ -104,14 +123,14 @@ func All(ctx context.Context, clients ClientSet, apis api.APIs) error {
 		errs++
 	}
 
-	if reflect.ValueOf(clients.Automation).IsNil() {
+	if clients.Automation == nil {
 		log.Warn("Skipped deletion of Automation configurations as API client was unavailable.")
 	} else if err := automation.DeleteAll(ctx, clients.Automation); err != nil {
```
Comment (UnseenWizzard) [inline pkg/delete/delete.go:128]: does automation.Delete() adhere to the ordering when it deletes things?

**Review by UnseenWizzard — APPROVED** [review]
lgtm

**Review by UnseenWizzard — APPROVED** [review]

**Pull request merged**